packages/camera/camera_avfoundation/ios/Classes/FLTCam.m

```diff
@@ -838,6 +861,55 @@ - (void)resumePreviewWithResult:(FLTThreadSafeFlutterResult *)result {
   [result sendSuccess];
 }
 
+- (void)setDescriptionWhileRecording:(NSString *)cameraName
+                              result:(FLTThreadSafeFlutterResult *)result {
+  if (!_isRecording) {
+    [result sendErrorWithCode:@"setDescriptionWhileRecording"
```

Comment:
it's odd to use function name as error code (i've never seen it, but not saying it's a bad idea). 

@stuartmorgan do we want to recommend developers to surface a Dart error using the "approach 2" in the [error handling proposal](https://docs.google.com/document/d/1Iqt36mp_MWX7Ou2l9tQpUPjlNbh0rSRrni7X0DZK3pk/edit?resourcekey=0-XdZ6l6uykkhbBbivPl64lg)? (It would be the very first Dart error in our 1P plugins so wanna to double check with you)

Reply:
Ill change it to setDescriptionWhileRecordingFailed and change the function name to setDescriptionWhileRecordingWithResult so it matches everything else